pkg/catalog/routes_test.go

```diff
@@ -937,6 +937,165 @@ func TestBuildInboundPoliciesSameNamespace(t *testing.T) {
 	assert.ElementsMatch(expectedPolicies, actual)
 }
 
+func TestBuildInboundAllowAllPolicies(t *testing.T) {
+	assert := tassert.New(t)
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	mockKubeController := k8s.NewMockController(mockCtrl)
+	mockMeshSpec := smi.NewMockMeshSpec(mockCtrl)
+	mockEndpointProvider := endpoint.NewMockProvider(mockCtrl)
+
+	mc := MeshCatalog{
+		kubeController:     mockKubeController,
+		meshSpec:           mockMeshSpec,
+		endpointsProviders: []endpoint.Provider{mockEndpointProvider},
+	}
+
+	bookstoreSA := service.K8sServiceAccount{
+		Name:      "bookstore",
+		Namespace: "bookstore-ns",
+	}
+
+	bookbuyerSA := service.K8sServiceAccount{
+		Name:      "bookbuyer",
+		Namespace: "bookbuyer-ns",
+	}
+	bookstoreMeshService := service.MeshService{
+		Name:      "bookstore",
+		Namespace: "bookstore-ns",
+	}
+	bookstoreK8sService := tests.NewServiceFixture(bookstoreMeshService.Name, bookstoreMeshService.Namespace, map[string]string{})
+
+	serviceAccounts := []*corev1.ServiceAccount{}
+	serviceAccounts = append(serviceAccounts, tests.NewServiceAccountFixture(tests.BookbuyerServiceAccountName, "bookbuyer-ns"))
+	serviceAccounts = append(serviceAccounts, tests.NewServiceAccountFixture(tests.BookstoreServiceAccountName, "bookstore-ns"))
+
+	mockEndpointProvider.EXPECT().GetID().Return("fake").AnyTimes()
+	mockKubeController.EXPECT().GetService(bookstoreMeshService).Return(bookstoreK8sService).AnyTimes()
+	mockKubeController.EXPECT().ListServiceAccounts().Return(serviceAccounts).AnyTimes()
+
+	expectedBookstoreHostnames := []string{
+		"bookstore",
+		"bookstore.bookstore-ns",
+		"bookstore.bookstore-ns.svc",
+		"bookstore.bookstore-ns.svc.cluster",
+		"bookstore.bookstore-ns.svc.cluster.local",
+		"bookstore:8888",
+		"bookstore.bookstore-ns:8888",
+		"bookstore.bookstore-ns.svc:8888",
+		"bookstore.bookstore-ns.svc.cluster:8888",
+		"bookstore.bookstore-ns.svc.cluster.local:8888",
+	}
+	bookstoreWeightedCluster := service.WeightedCluster{
+		ClusterName: "bookstore-ns/bookstore",
+		Weight:      100,
+	}
+
+	expectedPolicies := []*trafficpolicy.InboundTrafficPolicy{
+		{
+			Name:      "bookstore-bookstore-ns",
+			Hostnames: expectedBookstoreHostnames,
+			Rules: []*trafficpolicy.Rule{
+				{
+					Route: trafficpolicy.RouteWeightedClusters{
+						HTTPRouteMatch:   wildCardRouteMatch,
+						WeightedClusters: mapset.NewSet(bookstoreWeightedCluster),
+					},
+					AllowedServiceAccounts: mapset.NewSet(bookstoreSA, bookbuyerSA),
+				},
+			},
+		},
+	}
+	actual := mc.buildInboundAllowAllPolicies(bookstoreMeshService)
+	assert.Len(actual, 1)
+	assert.ElementsMatch(expectedPolicies, actual)
+}
+
+func TestBuildOutboundAllowAllPolicies(t *testing.T) {
```

Comment:
Same comment as L974. I'd like for this to be a table driven test to be able to easily add new test inputs.

Reply:
fixed